libexec/revng/revng-check-conventions

```diff
@@ -488,6 +503,62 @@ class CPPCheckPass(Pass):
         )
 
 
+class CheckCommitPass(Pass):
+    def __init__(self):
+        super().__init__("check-commit")
+
+    async def run(
+        self, files: Dict[str, Dict[Path, str]], args: Dict[str, str | bool]
+    ) -> PassResult:
+        if not args["check_commit"]:
+            return PassResult("", True)
+
+        commit_msg_cmd = await run_with_capture(["git", "log", "--pretty=%B", "HEAD^..HEAD"])
+        if commit_msg_cmd.returncode != 0:
+            return PassResult(commit_msg_cmd.output, False)
+
+        parts = commit_msg_cmd.output.splitlines()
+
+        if len(parts) < 3:
+            return PassResult("Missing commit message body", False)
+
+        if parts[1] != "":
+            return PassResult("Subject not divided by an empty line", False)
+
+        subject = parts.pop(0)
+
+        first_non_link = False
+        message_lines: List[str] = []
+        for line in reversed(parts):
+            if not first_non_link:
+                if re.match(r"\[\d+\] ", line) or line.strip() == "":
+                    continue
+
+                first_non_link = True
+            message_lines.insert(0, line)
+
+        problems = []
+        # Actual content checks
+        if len(subject) > 50:
+            problems.append("Commit subject line exceeds 50 characters")
+
```

Comment:
1. If there's more than one line, the second one should be blank.
2. Ensure no trailing whitespaces

Reply:
3. First line should *not* end with a `.`
4. If there are more than two lines, the last character of the commit must be `.`

Reply:
For (4) do you mean (a) If there is a commit description (e.g. subject + blank line + description) it must end with a `.` or (b) that if the commit description is longer than 2 lines it must end with a `.`?

Reply:
Basically the 72-chars-wrapped part, if present, must end with `.`.

Reply:
ACK